src/main/java/main/controllers/Project/SuiteController.java

```diff
@@ -86,4 +90,60 @@ public boolean delete(TestSuiteDto template) throws AqualityException {
         }
         return testSuites;
     }
+
+    public void syncLegacyTests(List<TestDto> legacyTests, Integer suiteId, boolean removeNotExecutedResults) throws AqualityException {
+        if(baseUser.isManager() || baseUser.getProjectUserBySuiteId(suiteId).isManager()){
+            legacyTests.forEach(test -> {
+                Test2SuiteDto test2Suite = new Test2SuiteDto();
+                test2Suite.setTest_id(test.getId());
+                test2Suite.setSuite_id(suiteId);
+                try {
+                    test2SuiteDao.delete(test2Suite);
+                    if(removeNotExecutedResults) {
+                        removePendingResultsForTest(suiteId, test.getId());
+                    }
+                } catch (AqualityException e) {
+                    e.printStackTrace();
+                }
+            });
+        }else{
+            throw new AqualityPermissionsException("Account is not allowed to Sync Test Suite", baseUser);
+        }
+    }
+
+    private void removePendingResultsForTest(Integer suiteId, Integer testId) throws AqualityException {
+        List<TestResultDto> testResults = testResultDao.selectSuiteLegacyResults(suiteId, testId);
+        testResults.forEach(testResult -> {
+            try {
+                testResultDao.delete(testResult);
+            } catch (AqualityException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    public List<TestDto> findLegacyTests(Integer suiteId, Integer notExecutedFor) throws AqualityException {
+        if (baseUser.isManager() || baseUser.getProjectUserBySuiteId(suiteId).isManager()) {
```

Comment:
same as for syncLegacyTests, are you sure that manager will do it?

Reply:
fixed